src/ameba/rule/style/redundant_self.cr

```diff
@@ -0,0 +1,110 @@
+require "compiler/crystal/syntax/token"
+
+module Ameba::Rule::Style
```

Comment:
I’m wondering if this would be better placed under Lint group, since it’s a code smell rather than a style issue.

Reply:
Personally I would lean on the side of `Style` group, also because of how RuboCop handles it.